guardian/newsletters-nx
# refactor: tidy up ui routes and remove /v1 from api

## What does this change?



- Removes `/v1` prefix because we are starting from scratch so let's not bother with API versioning as it opens us up to hell down the line! We were also already seeing very inconsistent usage of the prefix which is confusing for everyone
  - Have renamed the old "legacy" endpoint `GET /legacy/newsletters` but we can call it what we want - it just needs to be obviously the legacy version and not the new one
- Removes old code and duplicated options from the home page and menu bar
- Combines newsletter-related routes (including drafts)
- Moves `/wizard` and `/forms` into a `/demo` route which is intended for development purposes (and not quite ready for use?)
- Provides more intuitive UI
- Renamed product for fun (I can revert that, it was only silly!)

## How to test



Run locally (`npm run dev`) and have a click around - everything should still work on the API and UI.

## Images



<img width="1007" alt="image" src="https://user-images.githubusercontent.com/43961396/222224247-9eed0425-31bc-4e48-ae92-74fc0d56f4d2.png">

![image](https://user-images.githubusercontent.com/43961396/222224300-811026b3-9202-4c74-84d2-db9761363bbb.png)

![image](https://user-images.githubusercontent.com/43961396/222224346-12876d67-05d8-452c-a9ef-032f9d8c2ba3.png)

![image](https://user-images.githubusercontent.com/43961396/222224386-43575e2a-3f65-4351-b9ef-04de675092b9.png)


## What next?

Wonder what we should do with the `/` home route hmmm (maybe just auto redirect to `/newsletters`?)
